### PR TITLE
Feature/compile expression

### PR DESCRIPTION
**Pull request by hihiboss**
resolved: 

details:

1. Added some functions for compiling expression.
2. Implemented Emit() function in bytecode.
3. Implemented compileBooleanLiteral with test cases.

- [x] Test case

### CODE REVIEW

File: translate/compiler.go [hunk 1]
```diff
@@ -16,30 +16,41 @@
...
 		RawByte: make([]byte, 0),
 		AsmCode: make([]string, 0),
+		Abi: abi.ABI{
+			Methods: make([]abi.Method, 0),
+		},
 	}
 
 	for _, f := range c.Functions {
 		if err := compileFunction(*f, bytecode); err != nil {
 			return *bytecode, err
```
Comment (boohyunsik) [inline translate/compiler.go:42]: Why return bytecode also?

File: translate/compiler_internal_test.go [hunk 2]
```diff
@@ -16,7 +16,24 @@
...
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/DE-labtory/koa/abi"
+	"github.com/DE-labtory/koa/ast"
+)
+
+// TODO: implement test cases :-)
+func Test_generateFuncJumper(t *testing.T) {
```
Comment (boohyunsik) [inline translate/compiler_internal_test.go]: Hmm we need to talk about test function's name.
In parser, we named like TestGenerateFuncJumper()

File: translate/compiler_internal_test.go [hunk 3]
```diff
@@ -29,6 +46,125 @@ func Test_compileStatement(t *testing.T) {
...
+	for i, test := range tests {
+		bytecode := &Bytecode{
+			RawByte: make([]byte, 0),
+			AsmCode: make([]string, 0),
+			Abi:     abi.ABI{},
+		}
+		err := compileBooleanLiteral(test.expression, bytecode)
+
+		if err != nil {
+			t.Fatalf("test[%d] - compileBooleanLiteral() had error. err=%v", i, err)
```
Comment (boohyunsik) [inline translate/compiler_internal_test.go:145]: We need to test whether function returns valid error or not.

**Review by boohyunsik — CHANGES_REQUESTED** [review]

File: translate/compiler.go (hunk 1 above)
Comment (hihiboss) [reply translate/compiler.go:42]: @boohyunsik 
Your question means if error occurs, why return `*bytecode, err`, not `Bytecode{}, err`??
To be easier to debug or test, it returns the compiled bytecode so far.

File: translate/compiler_internal_test.go (hunk 2 above)
Comment (hihiboss) [reply translate/compiler_internal_test.go]: Oh, i'll modify it!

File: translate/compiler_internal_test.go (hunk 3 above)
Comment (hihiboss) [reply translate/compiler_internal_test.go:145]: But, the error is came out from just `encoding.EncodeOperand()`  not `compileBooleanLiteral()`. So, I think that's enough... Do you think it need?

Comment (boohyunsik) [reply translate/compiler_internal_test.go:145]: Anyway, the way for higher coverage is good. I think.

File: translate/compiler.go [hunk 4]
```diff
@@ -106,3 +139,93 @@ func compileIfStatement(s *ast.IfStatement, bytecode *Bytecode) error {
...
+
+	case *ast.Identifier:
+		return compileIdentifier(expr, bytecode)
+
+	case *ast.ParameterLiteral:
+		return compileParameterLiteral(expr, bytecode)
+
+	default:
+		return nil
+	}
```
Comment (junbeomlee) [inline translate/compiler.go:186]: should return errors when default!
```
default:
   return errors.New("error occurred")
```

**Comment by AgwaB** [comment]
LGTM but you implement `Emerge` not `Emit` right?

**Comment by zeroFruit** [comment]
After feedback the reviews which commented so far, looks good to me

**Comment by hihiboss** [comment]
@leesd556 
Yes, just changed the name!
Emit means that function changes parameter like `Emit(bytecode)`
But bytecode will have that function, Emerge would be better. Like `Bytecode.Emerge()`

File: translate/compiler_internal_test.go (hunk 3 above)
Comment (junbeomlee) [reply translate/compiler_internal_test.go:145]: we can not mock encoding function so i think this test case is enough!

**Comment by junbeomlee** [comment]
Did u decide to add tracer?

**Pull request merged**